core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1583,6 +1583,108 @@ public static function getDriverForTable(string $table): string
 
 		return $dataContainer;
 	}
+
+	/**
+	 * Generates the label for a given data record according to the DCA configuration.
+	 *
+	 * @param array  $row   The data record
+	 * @param string $table The name of the data container
+	 * @param array  $args  The generated labels for each column
+	 */
+	public static function generateRecordLabel(array $row, string $table, array &$args = array()): string
+	{
+		$mode = $GLOBALS['TL_DCA'][$table]['list']['sorting']['mode'] ?? self::MODE_SORTED;
+		$showFields = $GLOBALS['TL_DCA'][$table]['list']['label']['fields'];
+
+		// Label
+		foreach ($showFields as $k=>$v)
+		{
+			// Decrypt the value
+			if ($GLOBALS['TL_DCA'][$table]['fields'][$v]['eval']['encrypt'] ?? null)
+			{
+				$row[$v] = Encryption::decrypt(StringUtil::deserialize($row[$v]));
+			}
+
+			if (strpos($v, ':') !== false)
+			{
+				list($strKey, $strTable) = explode(':', $v, 2);
+				list($strTable, $strField) = explode('.', $strTable, 2);
+
+				$objRef = Database::getInstance()
+					->prepare("SELECT " . Database::quoteIdentifier($strField) . " FROM " . $strTable . " WHERE id=?")
+					->limit(1)
+					->execute($row[$strKey]);
+
+				$args[$k] = $objRef->numRows ? $objRef->$strField : '';
+			}
+			elseif (\in_array($GLOBALS['TL_DCA'][$table]['fields'][$v]['flag'] ?? null, array(self::SORT_DAY_ASC, self::SORT_DAY_DESC, self::SORT_MONTH_ASC, self::SORT_MONTH_DESC, self::SORT_YEAR_ASC, self::SORT_YEAR_DESC)))
+			{
+				if (($GLOBALS['TL_DCA'][$table]['fields'][$v]['eval']['rgxp'] ?? null) == 'date')
+				{
+					$args[$k] = $row[$v] ? Date::parse(Config::get('dateFormat'), $row[$v]) : '-';
+				}
+				elseif (($GLOBALS['TL_DCA'][$table]['fields'][$v]['eval']['rgxp'] ?? null) == 'time')
+				{
+					$args[$k] = $row[$v] ? Date::parse(Config::get('timeFormat'), $row[$v]) : '-';
+				}
+				else
+				{
+					$args[$k] = $row[$v] ? Date::parse(Config::get('datimFormat'), $row[$v]) : '-';
+				}
+			}
+			elseif (($GLOBALS['TL_DCA'][$table]['fields'][$v]['eval']['isBoolean'] ?? null) || (($GLOBALS['TL_DCA'][$table]['fields'][$v]['inputType'] ?? null) == 'checkbox' && !($GLOBALS['TL_DCA'][$table]['fields'][$v]['eval']['multiple'] ?? null)))
+			{
+				$args[$k] = $row[$v] ? $GLOBALS['TL_LANG']['MSC']['yes'] : $GLOBALS['TL_LANG']['MSC']['no'];
```

Comment:
This has been handled differently in tree view before:

https://github.com/contao/contao/blob/6e3156a0703917cd137d26ecbe3765130aed8276/core-bundle/src/Resources/contao/drivers/DC_Table.php#L4085-L4088

Did you intentionally change this?

Reply:
Not intentionally, no. But which version is more correct? Probably the one from the list view? (i.e. the one currently implemented in the method)

Reply:
IDK. Can you post a screenshot of how the change affects the tree view labels?

Reply:
The core does not use tree views with boolean fields in their label. But you if you add

````php
// contao/dca/tl_page.php
$GLOBALS['TL_DCA']['tl_page']['list']['label']['fields'] = ['title', 'published'];
$GLOBALS['TL_DCA']['tl_page']['list']['label']['format'] = '%s <span style="opacity:0.5;font-size:0.8em;">published: %s</span>';
```

then this is what it looks like now:

![image](https://user-images.githubusercontent.com/4970961/148758226-9f01369a-9335-4424-9de5-62cb5e352048.png)

and this is what it looks like with the current state of the PR:

![image](https://user-images.githubusercontent.com/4970961/148758178-3ee7b6a8-c768-4918-b3a2-5d25ba1158ab.png)

I can of course check the mode and make sure that in the tree view it will show the field's label when true and not yes/no.

Reply:
> Though I think this should be consistent across all views.

I agree. 👍